File: clojure/rna-transcription/src/rna_transcription.clj
(ns rna-transcription)

(defn to-rna [dna]
  (->> dna
    (map #(condp = %
            \G \C
            \C \G
            \T \A
            \A \U
            (assert false (str "Invalid nucleotide " %))))
    (apply str "")))

File: typescript/rna-transcription/rna-transcription.ts
function toComplement(nucleotide: string): string {
  switch (nucleotide) {
      case 'C': return 'G'
      case 'G': return 'C'
      case 'A': return 'U'
      case 'T': return 'A'
      default: throw Error('Invalid input DNA.')
  }
}

export function toRna(dna: string): string {
  return dna.split('').map(toComplement).join('');
}
